src/Operations.jl

```diff
@@ -1282,6 +1282,7 @@ function gen_test_code(testfile::String;
         --color=$(Base.have_color === nothing ? "auto" : Base.have_color ? "yes" : "no")
         --compiled-modules=$(Bool(Base.JLOptions().use_compiled_modules) ? "yes" : "no")
         --check-bounds=yes
+        --depwarn=yes
```

Comment:
This makes `Pkg.test` not to propagate `--depwarn=error`. So I think we need to look at `JLOptions()` and try not to "downgrade" `depwarn` setting.

Reply:
Hmm yes, on balance I think that's probably the most consistent with the other options. I added a tweak.